### ISSUE TITLE
CSV.read `multithreadpostparse`: erroring 

### ISSUE THREAD

**Issue by anandijain** [JuliaData/CSV.jl#1029]
I have a repo that should be a working reproducer, though it takes quite a while (and slams wikipedia :P).

https://github.com/anandijain/WikipediaDiGraph.jl/blob/main/get_data.sh
https://github.com/anandijain/WikipediaDiGraph.jl/blob/main/test/runtests.jl

I was able to avoid the issue with `ntasks=1` on the last CSV.read call, but it takes considerably longer.
```julia
kws = (escapechar='\\', quotechar=''', silencewarnings=true)
df = CSV.read(dst, DataFrame; types=Union{String, Missing}, kws...)
```
gives 
`nested task error: TypeError: in typeassert, expected SentinelArrays.SentinelVector{String15, String15, Missing, Vector{String15}}, got a value of type SentinelArrays.SentinelVector{Int64, Int64, Missing, Vector{Int64}}`

unfortunately I don't have the stacktrace anymore

**Comment by quinnj** [comment]
Fixed in CSV.jl v1.0.0. Here's the full triage, since this one needed a specific combination to reproduce.

### What actually triggers it

Two things have to happen together:

1. **Some data rows carry more fields than the header.** In 0.10.x, a parse task that sees an extra field pushes its own task-local `Column` straight into the shared top-level `columns` vector (`multithreadparse` in `src/file.jl`), so the widened column is aliased between that task and the parent.
2. **That widened column parses as `Int64` in some byte chunks and as a short string in others.** `multithreadpostparse` then has to promote every chunk to the final string type, and for a pool-eligible column `checkpooled!` asserts `SVec{String15}` but gets the un-promoted `SVec{Int64}`.

Ingredient 1 is exactly what the reproducer script produces, and it isn't obvious from reading it. In `WikipediaDiGraph.jl/test/runtests.jl`, `clean_page_line` does `split(c, "NULL),")`, which splits the MySQL `INSERT` line on the literal text `NULL),`. Any record whose `page_lang` is *not* NULL therefore doesn't get split, gets merged with the following record, and the emitted CSV row ends up with extra fields. Ingredient 2 comes from `page_links_updated`: a quoted 14-digit number (`Int64`) in most chunks, and the bare literal `NULL` in others, which alongside the 14-char timestamps promotes to `String15`.

Neither ingredient alone reproduces it. 60 well-formed files with cross-chunk type flips parsed cleanly on v0.10.4.

### Reproduced

Generated ragged CSVs (random column-type patterns, with every *n*-th row carrying 1–3 extra fields), read once in parallel and once with `ntasks=1`, then compared size / eltype / values:

| corpus | v0.10.4 (Julia 1.8.5) | v0.10.17 (Julia 1.12.6) | **v1.0.0 (Julia 1.12.6)** |
|---|---|---|---|
| ragged, 110 seeds | 17 / 40 failed | **14 / 110 failed** | **0 / 110 failed** |
| well-formed, 60 seeds | 0 / 60 | — | 0 / 60 |
| real `simplewiki` page dump (947,771 rows, 102 MB) | ok | ok | ok |

10 threads throughout. The v0.10.17 failure is this issue's message, character for character:

```
nested task error: TypeError: in typeassert,
  expected SentinelArrays.SentinelVector{String15, String15, Missing, Vector{String15}},
  got a value of type SentinelArrays.SentinelVector{Int64, Int64, Missing, Vector{Int64}}
Stacktrace:
 [1] checkpooled!(::Type{Union{Missing, String15}}, pertaskcolumns::Vector{Vector{CSV.Column}}, ...)
```

So this was never fixed on the 0.10.x line — it still reproduces on v0.10.17. One seed lands instead on `UndefVarError: T not defined in static parameter matching` from `SentinelArrays.ChainedVector`, same root cause (an un-promoted chunk column), different landing spot.

### Why v1.0.0 can't hit it

The 1.0 rewrite (#1196) removes the machinery entirely: `multithreadpostparse`, `promotetostring!` and the per-task `Vector{Vector{Column}}` promotion no longer exist in `src/`. Type resolution is a single shared promotion state with a convergence loop, and the column set is fixed by the header. From `docs/src/migration.md`:

> a long row does not widen the schema; extra fields add a problem

That makes ingredient 1 structurally impossible. Extra fields are reported rather than silently widening the table:

```
CSV.Problem(long_row at data row 1009, column 0, byte 39156:
            "expected 4 fields, found 7 (extra fields ignored)")
```

### Verification

The original 2022 enwiki dump is no longer on the dumps server, so I ran the reproducer's own pipeline (`clean_page_line` → `page_lines.csv`) over `simplewiki-latest-page.sql` (103 MB, 947,771 rows, 12 columns, genuinely ragged — 38,914 over-long rows). SHA-256 per column: **all 12 columns identical** across v0.10.17 serial, v0.10.17 parallel, v1.0.0 serial and v1.0.0 parallel.

Comparing v0.10.17-serial against v1.0.0-parallel across the fuzz corpus, every shared column matched except integer columns containing `typemin(Int64)+1`, where **v0.10.17 is the incorrect one**: it reads `-9223372036854775807` as `+9223372036854775807` (26,006 / 26,006 occurrences in one file — the sign is dropped). v1.0.0 reads it correctly; this looks like the Parsers 2.x sign-after-prefix bug fixed in Parsers 3.

### One note for anyone hitting this on their own data

Under 1.0 a long row no longer widens the schema, so a `page_lines.csv` built by the script above will now give you the header's columns plus a `:long_row` entry in `CSV.problems(f)` for each merged record, instead of silently gaining extra columns. The `split(line, "NULL),")` cleanup is the actual defect there and is worth fixing independently — splitting on `),(` or parsing the tuple list properly would avoid merging records.

Closing as fixed by #1196.

---

*[issue investigated and resolved by AI; reviewed by quinnj; please respond with any discrpancies or inaccuracies]*
